docs/docs/troubleshooting.md

```diff
@@ -0,0 +1,165 @@
+
+The resolution is to flush the Redis database with [`FLUSHDB`](https://redis.io/commands/flushdb) or [`FLUSHALL`](https://redis.io/commands/FLUSHALL).
+
+### RPC Errors
+
+#### certificate signed by unknown authority
+
+When authenticating and authorizing a user for the first time, you may see the following in your Pomerium logs:
+
+```log
+ERR http-error error="401 Unauthorized: ..... rpc error: code = DeadlineExceeded desc = latest connection error: connection error: desc = "transport: authentication handshake failed: x509: certificate signed by unknown authority...."
+```
+
+**Why**
+
+This error means that the proxy is rejecting the authorize service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that is associated with the supplied certificate. This can happen for a few reasons.
+
+**Solution**
+
+Ensure that the proxy service knows about and trusts the certificate authority that signed the authorize service's certificate.
+
+- Add the certificate authority directly into Pomerium using the [certificate authority](/reference/readme.md#certificate-authority) config setting.
+- Add the certificate authority to the system's underlying trust store.
+- Replace your system / docker image certificate bundle.
+
+    For Docker:
+
+    ```docker
+    COPY --from=builder /etc/ssl/certs/your-cert-bundle.crt /etc/ssl/certs/ca-certificates.crt
+    ```
+- Finally, ensure that you aren't being man-in-the-middle'd or that some eager router isn't injecting its own certificate along the way. Use openssl to verify that your proxy service is getting the certificate you think its getting.
+
+    ```bash
+    openssl s_client -servername pomerium.io -connect pomerium.io:443 </dev/null \
+    | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p'
+    ```
+
+#### rpc error: code = DeadlineExceeded
+
+When authenticating and authorizing a user for the first time, you may get the following in your Pomerium logs.
+
+```log
+{"level":"error",..."error":"rpc error: code = DeadlineExceeded desc = context deadline exceeded","http-message":"rpc error: code = DeadlineExceeded desc = context deadline exceeded","http-code":500,"message":"http-error"}
+```
+
+**Why**
+
+The proxy service is not able to create a connection with the authorization service to validate a user.
+
+**Solution**
+
+Usually, this is the result of either a routing issue or a configuration error. Make sure that you are using the _internally_ routable URL for authorize service. Many cloud loud balancers _do not_ yet support gRPC transposing the ingress. So while your authenticate service url will probably look like `https://authenticate.corp.example.com`, your authorizer service url will likely be more like `https://pomerium-authorize-service.default.svc.cluster.local` or `https://localhost:5443`.
```

Comment:
```suggestion
Usually, this is the result of either a routing issue or a configuration error. Make sure that you are using the _internally_ routable URL for authorizing the service. Many cloud loud balancers _do not_ yet support gRPC transposing the ingress. So while your authenticate service url will probably look like `https://authenticate.corp.example.com`, your authorizer service url will likely be more like `https://pomerium-authorize-service.default.svc.cluster.local` or `https://localhost:5443`.
```
Should "ingress" be capitalized in this?

Reply:
As discussed above, authorize is a noun not an adjective in this context. But the grammar was off either way. I copied this from the old FAQ page and did not review/rewrite it.
```suggestion
Usually, this is the result of either a routing issue or a configuration error. Make sure that you are using the _internally_ routable URL for the authorize service. Many cloud loud balancers _do not_ yet support gRPC transposing the ingress. So while your authenticate service url will probably look like `https://authenticate.corp.example.com`, your authorizer service url will likely be more like `https://pomerium-authorize-service.default.svc.cluster.local` or `https://localhost:5443`.
```

re "ingress": in this context we're not explicitly talking about a Kubernetes object type, so we're not bound by their... questionable choices in capitalization.